src/state-summary/state-card-select.ts

```diff
@@ -0,0 +1,99 @@
+import "@polymer/paper-dropdown-menu/paper-dropdown-menu-light";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-listbox/paper-listbox";
+import {
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+} from "lit";
+import { customElement, property } from "lit/decorators";
+import { stopPropagation } from "../common/dom/stop_propagation";
+import { computeStateName } from "../common/entity/compute_state_name";
+import "../components/entity/state-badge";
+import { SelectEntity, setSelectOption } from "../data/select";
+import type { HomeAssistant } from "../types";
+
+@customElement("state-card-select")
+class StateCardSelect extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property() public stateObj!: SelectEntity;
+
+  protected render(): TemplateResult {
+    return html`
+      <state-badge .stateObj=${this.stateObj}></state-badge>
+      <paper-dropdown-menu-light
+        .label=${computeStateName(this.stateObj)}
+        @iron-select=${this._selectedOptionChanged}
+        @click=${stopPropagation}
+      >
+        <paper-listbox slot="dropdown-content">
+          ${this.stateObj.attributes.options.map(
+            (option) =>
+              html`
+                <paper-item .option=${option}
+                  >${(this.stateObj.attributes.device_class &&
+                    this.hass.localize(
+                      `component.select.state.${this.stateObj.attributes.device_class}.${option}`
+                    )) ||
+                  this.hass.localize(`component.select.state._.${option}`) ||
+                  option}</paper-item
+                >
+              `
+          )}
+        </paper-listbox>
+      </paper-dropdown-menu-light>
+    `;
+  }
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+    if (!changedProps.has("stateObj")) {
+      return;
+    }
+    // Update selected after rendering the items or else it won't work in Firefox
+    this.shadowRoot!.querySelector(
+      "paper-listbox"
+    )!.selected = this.stateObj.attributes.options.indexOf(this.stateObj.state);
+  }
+
+  private async _selectedOptionChanged(ev) {
+    const option = ev.target.selectedItem.option;
+    if (option === this.stateObj.state) {
+      return;
+    }
+    setSelectOption(this.hass, this.stateObj.entity_id, option);
```

Comment:
This will help with stack traces:
```suggestion
    await setSelectOption(this.hass, this.stateObj.entity_id, option);
```

Reply:
This was removed by @bramkragten in <https://github.com/home-assistant/frontend/pull/9422#pullrequestreview-686712532>

Reply:
Alright 👍